scipy/stats/stats.py

```diff
@@ -3445,9 +3454,23 @@ def kendalltau(x, y, initial_lexsort=True):
 
     KendalltauResult = namedtuple('KendalltauResult', ('correlation', 'pvalue'))
 
-    if not x.size or not y.size:
+    if x.size != y.size:
+        raise ValueError('All inputs must be of the same size')
```

Comment:
Maybe help the user a bit more here:

    All inputs to `kendalltau` must be of the same size,
    found x-size %s and y-size %s" % (x.size, y.size)"